core/src/main/scala/org/apache/gearpump/metrics/Metrics.scala

```diff
@@ -48,6 +51,15 @@ object Metrics extends ExtensionId[Metrics] with ExtensionIdProvider {
   val LOG: Logger = LogUtil.getLogger(getClass)
   import org.apache.gearpump.util.Constants._
 
+  sealed trait MetricType
+  case class Histogram(count: Long, min: Long, max: Long, mean: Double, stddev: Double, median: Double, p75: Double, p95: Double, p98: Double, p99: Double, p999: Double) extends MetricType
```

Comment:
We may need javascript class's which uPickle provides. Let's handle in subsequent task.